python/triton/knobs.py

```diff
@@ -169,8 +170,15 @@ class NvidiaTool:
     path: str
     version: str
 
+    @classmethod
+    def from_path(cls, path: str) -> Optional[NvidiaTool]:
+        return cls._from_path(path, os.getenv("PATH"))
```

Comment:
I looked at the history and the code this is derived from was indeed cached, and didn't even bother with the defensiveness around `PATH`.  So I would say it's probably not needed.

https://github.com/triton-lang/triton/blob/aaa9932a8bc04cde0304d5c87820837b2cf10de8/third_party/nvidia/backend/compiler.py#L33-L48